src/kaleidoscope/hardware/keyboardio/Model01.h

```diff
@@ -43,11 +44,20 @@ class Model01 : public kaleidoscope::Hardware {
   static constexpr byte matrix_columns = 16;
   static constexpr int8_t led_count = 64;
 
+  typedef MatrixAddr<matrix_rows, matrix_columns> KeyAddr;
+  typedef MatrixAddr<matrix_rows, matrix_columns> KeyLEDAddr;
```

Comment:
I just realized that there's something here that concerns me: `KeyAddr` and `KeyLEDAddr` for the Model01 are the same type, even though they appear to be different types in use (and, for other boards, they are different types). This means that it's not possible to have two polymorphic functions like:
```
int foo(KeyAddr key_addr);
int foo(KeyLEDAddr led_addr);
```
…and it's possible to accidentally pass one type where the other was intended, and get  unexpected results without a compilation error.

Reply:
This question is connected to the question raised by @obra above, about what a `KeyAddr` or `KeyLEDAddr` actually represents. For me `KeyAddr`s and `KeyLEDAddr`s are both unique ids that the user can use to reference keys or a key-LEDs. The way it is currently implemented, a `KeyAddr` and a `KeyLEDAddr` have identical values if they both reference the same keys. That's not because the two C++ types are identical but because they both replaced the same row/col-pairs. But this very identity raises the question whether we actually need two sets of addresses at all. 

i had introduced `LEDAddr` because i initially thought there was just one set of LEDs that don't necessarily need to be per-key LEDs. Then, obra reminded me that there will be other sets of LEDs coming, like underglow or status LEDs. Because of that i renamed `LEDAddr` to `KeyLEDAddr`. 

My conclusion is: We should go with only a single `KeyAddr` for both keys and key related LEDs. The wiring details should be entirely handled by the hardware classes. For other sets of LEDs we should have additional address-classes like `UnderglowLEDAddr` or `StatusLEDAddr` that we introduce once we need them.

Reply:
Ah, I see. I fell into the trap of assuming that your `KeyLEDAddr` was equivalent to my `LedAddr`, but it's not. I agree with your conclusion — if the same value refers to the same physical position on the keyboard in both cases, it would be better to use just the one name.